be/src/exec/pipeline/pipeline_driver.h

```diff
@@ -440,7 +463,27 @@ class PipelineDriver {
     void set_driver_queue_level(size_t driver_queue_level) { _driver_queue_level = driver_queue_level; }
 
     inline bool is_in_ready_queue() const { return _in_ready_queue.load(std::memory_order_acquire); }
-    void set_in_ready_queue(bool v) { _in_ready_queue.store(v, std::memory_order_release); }
+    void set_in_ready_queue(bool v) {
+        SCHEDULE_CHECK(!v || !is_in_ready_queue());
+        _in_ready_queue.store(v, std::memory_order_release);
+    }
+
+    bool is_in_block_queue() const { return _in_block_queue.load(std::memory_order_acquire); }
+    void set_in_block_queue(bool v) {
+        SCHEDULE_CHECK(!v || !is_in_block_queue());
+        SCHEDULE_CHECK(!is_in_ready_queue());
+        _in_block_queue.store(v, std::memory_order_release);
+    }
+
+    ScheduleToken acquire_schedule_token() {
+        bool val = false;
+        return {this, _schedule_token.compare_exchange_strong(val, true)};
```

Comment:
ScheduleToken, what is the motivation to introduce it?   in order to prevent a driver triggered for multiple times simultaneously to cause others starving? add some verbose  explanation on it?

Reply:
I will add some comments.
This one is because there is a competition between event_scheduler->add_to_block_queue and observer.

If event_scheduler doesn't get the token, it proves that the observer has entered the critical zone, and we don't know the real state of the driver, so we need to try scheduling the driver one more time.